### PR TITLE
docs(harnesses): ensure all harnesses have complete and correct docs

### PR DESCRIPTION
**Pull request by mmalerba**

### CODE REVIEW

File: src/material/dialog/testing/dialog-harness.ts [hunk 1]
```diff
@@ -11,56 +11,57 @@ import {DialogRole} from '@angular/material/dialog';
...
+   * @param options Options for filtering which dialog instances are considered a match.
    * @return a `HarnessPredicate` configured with the given options.
    */
   static with(options: DialogHarnessFilters = {}): HarnessPredicate<MatDialogHarness> {
     return new HarnessPredicate(MatDialogHarness, options);
   }
 
-  /** Gets the id of the dialog. */
+  /** Gets a promise for the id of the dialog. */
   async getId(): Promise<string|null> {
```
Comment (devversion) [inline src/material/dialog/testing/dialog-harness.ts]: I'm somehow (personally) not happy about the way we need to repeat `promise for` all the time 😞 I've been following the concept of just saying e.g. `Gets the text of the tab content` The JSDoc method description does not necessarily refer to return value _type_.

I think we need to decide on how we want to document such methods in general. i.e. if we want to mention `Promise` each time, and what consistent pattern we should follow (like `Gets a promise for xxx`).

As an example: https://jsdoc.app/tags-returns.html. The function w/ promise does not mention promise in the function description. (`@returns` is equal to the type in TS)

Comment (devversion) [reply src/material/dialog/testing/dialog-harness.ts]: I'm not trying to block this PR going forward. I just thought about this too and would love to discuss this at some point (maybe we could even decide on a consistent format for future asynchronous methods)

Comment (mmalerba) [reply src/material/dialog/testing/dialog-harness.ts]: I could also see just not mentioning the fact that its a promise, but we should be consistent about it. I noticed that prior to this PR it was a mix of comments that mention it and ones that don't, so I chose to change them all to mention it.

Comment (devversion) [reply src/material/dialog/testing/dialog-harness.ts]: Yes. agreed. At the moment it is inconsistent, so this PR is definitely good! We can revisit this in the future. I'd prefer not mentioning promise, but we can discuss this together and made changes based on the outcome.

Comment (jelbourn) [reply src/material/dialog/testing/dialog-harness.ts]: This seems like a decent bikeshed to paint during the team meeting.

Comment (jelbourn) [reply src/material/dialog/testing/dialog-harness.ts]: (my opinion would be to just include the word "asynchronously" in the description, e.g. "Get the name of the button asynchronously").

File: src/material/dialog/testing/dialog-harness.ts [hunk 2]
```diff
@@ -11,56 +11,57 @@ import {DialogRole} from '@angular/material/dialog';
...
   async getAriaDescribedby(): Promise<string|null> {
     return (await this.host()).getAttribute('aria-describedby');
   }
 
   /**
-   * Closes the dialog by pressing escape. Note that this method cannot
-   * be used if "disableClose" has been set to true for the dialog.
+   * Closes the dialog by pressing escape.
+   *
+   * Note: this method cannot be used if `disableClose` has been set to `true` for the dialog.
```
Comment (jelbourn) [inline src/material/dialog/testing/dialog-harness.ts]: This says "cannot be used", but I would instead say something like "Does nothing if the `disableClose` option has been set for the dialog"

File: src/material/menu/testing/menu-harness.ts [hunk 3]
```diff
@@ -40,38 +40,57 @@ export class MatMenuHarness extends ComponentHarness {
...
 
+  /** Gets a promise for the text of the menu's trigger element. */
   async getTriggerText(): Promise<string> {
     return (await this.host()).text();
   }
 
-  /** Focuses the menu and returns a void promise that indicates when the action is complete. */
+  /**
+   * Focuses the menu
+   * @return A promise that resolves when the action is complete.
```
Comment (jelbourn) [inline src/material/menu/testing/menu-harness.ts]: I wonder if we could just update our dgeni setup to add this return automatically for `async` functions when the `@return` is left blank.

cc @devversion

Comment (devversion) [reply src/material/menu/testing/menu-harness.ts]: Yes. This is definitely possible. Here is a prototype/example: https://github.com/angular/components/pull/17751

**Comment by mmalerba** [comment]
@jelbourn removed unnecessary `@return` and mentions of "promise for", should be ready to go now.

**Pull request merged**